proposals/NNNN-SDL-device-listener.md

```diff
@@ -0,0 +1,244 @@
+
+## Motivation
+
+Due to the fact that the original server side bluetooth transport solution can't be altered, it's important that we try to mitigate the unnecessary starting of a foreground service. This is because when starting a foreground service, the application must include putting a notification into the top level notification drawer. An icon will show to the user, with a message that has been included in the library. The issue is that most users do not have SDL enabled hardware, so when they see this notification they wonder what it is and how to get rid of it.
+
+The SDL library already attempts to prevent the user from seeing the notification. This is done by quickly enters the foreground as required by the Android OS, then within 10 milliseconds, the service exits the foreground, notification removed, and notification channel deleted. However, based on testing different phones it was obvious that this has a multitude of different results. Some phones do exactly as expected and the notification is cleared before presented to the user. Other devices and OS versions will leave the notification in the tray for roughly 5 seconds until it is removed. The project maintainers have refactored the logic to avoid this for the majority of device and OS combinations, but some devices do not honor the APIs correctly. 
+
+Because of these adverse situations, the library should alter its logic that handles the starting of the `SdlRouterService`. This can be accomplished by creating a new gatekeeper class that prevents the service from ever being started until there has been a confirmed SDL connection via the standard SDL UUID. This logic can be inserted into the `SdlBroadcastReceiver` without a lot of overhead to existing code.
+
+## Proposed solution
+
+The solution will be to create a temporary bluetooth server socket in the broadcast receiver that receives the bluetooth connection intents. A new thread will be spawned off that will handle the listening. If a connection is made with the device, the bluetooth connection will be closed, and then the `SdlRouterService` will be started up directly to the foreground. A timeout handler will be present to keep the socket alive for a short duration without allowing it to run forever. Once the timeout expires, if there was no connection made, the socket will stop listening and the router service will not be started. In either case the MAC address will be saved into preferences with the results of a successful connection or not. 
```

Comment:
```suggestion
The solution will be to create a temporary bluetooth server socket in the broadcast receiver that receives the bluetooth connection intents. A new thread will be spawned off that which will handle the listening. If a connection is made with the device, the bluetooth connection will be closed, and then the `SdlRouterService` will be started up directly to the foreground. A timeout handler will be present to keep the socket alive for a short duration without allowing it to run forever. Once the timeout expires, if there was no connection made, the socket will stop listening and the router service will not be started. In either case the MAC address will be saved into preferences with the results of a successful connection or not. 
```

Reply:
>A new thread will be spawned off that will handle the listening.
Please update to "A new thread will be spawned off that which will handle the listening."